src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,83 @@
+package duke;
+import dukeUiElements.Ui;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.StandardOpenOption;
+
+/**
+ * This class deals with making sense of the user input and carrying out tasks accordingly.
+ */
+public class Parser {
+    public enum TaskKeyVal {ToDo, Deadline, Event, Delete, mark, unmark, bye, list, find};
+
+    /**
+     * Returns a boolean based on the user input choice.
+     * bye -> initiates user exit.
+     * list -> lists all the tasks.
+     * mark -> marks the task.
+     * unmark -> unmarks the task.
+     * ToDo -> creates a new ToDo and adds to task list.
+     * Deadline -> creates a new Deadline and adds to task list.
+     * Event -> creates a new Event and adds to task list.
+     * Delete -> Deletes task at given number.
+     *
+     * @param userInput String value provided to be made sense of and carry task.
+     * @return boolean value false to break out of program, true otherwise.
+     * @throws DukeException when incorrect / invalid input is entered.
+     */
+    public static String parse(String userInput) throws DukeException {
```

Comment:
Good to see that you have handled all the cases of inputs, you could also try using switch and case instead of if-else.